windows/guest_customization/check_autologon_count.yml

```diff
@@ -1,46 +1,50 @@
 # Copyright 2021-2023 VMware, Inc.
 # SPDX-License-Identifier: BSD-2-Clause
 ---
-# Get auto Admin logon value after gosc
-- include_tasks: ../utils/win_execute_cmd.yml
+# Check logon as Administrator automatically is set in guest OS
+- name: "Initialize the registry value of logon as Administrator automatically"
+  ansible.builtin.set_fact:
+    auto_admin_logon: 0
+
+- name: "Get logon as Administrator automatically registry value"
+  include_tasks: ../utils/win_execute_cmd.yml
   vars:
-    win_powershell_cmd: "(get-itemproperty -path 'HKLM:\\software\\microsoft\\Windows NT\\CurrentVersion\\Winlogon').AutoAdminLogon"
-- name: Set fact of the Admin auto logon
+    win_powershell_cmd: >-
+      (Get-ItemProperty -path 'HKLM:\\software\\microsoft\\Windows NT\\CurrentVersion\\Winlogon').AutoAdminLogon
+
+- name: "Set fact of logon as Administrator automatically registry value"
   ansible.builtin.set_fact:
     auto_admin_logon: "{{ win_powershell_cmd_output.stdout_lines[0] }}"
-- ansible.builtin.debug:
-    msg: "Get auto Admin logon enabled value: {{ auto_admin_logon }}"
-  when: enable_debug
+  when:
+    - win_powershell_cmd_output.stdout_lines is defined
+    - win_powershell_cmd_output.stdout_lines | length != 0
+
+- name: "Check logon as Administrator automatically is enabled"
+  ansible.builtin.assert:
+    that:
+      - "{{ auto_admin_logon | int == 1 }}"
+    success_msg: "Logon as Administrator automatically is enabled in guest OS."
+    fail_msg: "Logon as Administrator automatically registry value '{{ auto_admin_logon }}' is not equal to 1."
```

Comment:
Maybe it's better to also add message indicating Auto logon is disabled. 
Something like:
Logon as Administrator automatically is not enabled, the current registry value is '{{ auto_admin_logon }}', not expected value 1.

Reply:
Ok, thanks.